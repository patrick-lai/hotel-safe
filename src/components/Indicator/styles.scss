@import '../../theme/default.scss';

.indicator {
  background-color: $color-gray;
  border: 6px solid black;
  border-radius: 100%;
  width: 60px;
  height: 60px;
  transition: all 500ms ease;

  &.-error {
    background-color: $color-fail;
  }

  &.-success {
    background-color: $color-success;
  }
}
