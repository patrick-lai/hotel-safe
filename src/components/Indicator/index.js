/***
 * An indicator can can show status by color
 */

import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Indicator = ({ locked }) => (
  <div className={['indicator', locked ? '-error' : '-success'].join(' ')} />
);

Indicator.propTypes = {
  locked: PropTypes.bool
};

export default Indicator;
