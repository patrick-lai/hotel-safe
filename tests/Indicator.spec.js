/***
 * Indicator
 */

import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Indicator from '../src/components/Indicator';

Enzyme.configure({ adapter: new Adapter() });

describe('Indicator component', () => {
  test('should have correct classes on different states', () => {
    const locked = shallow(<Indicator locked={true} />);
    const open = shallow(<Indicator locked={false} />);
    expect(locked.html()).toMatchSnapshot();
    expect(open.html()).toMatchSnapshot();
  });
});
