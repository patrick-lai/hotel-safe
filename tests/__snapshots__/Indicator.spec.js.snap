// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator component should have correct classes on different states 1`] = `"<div class=\\"indicator -error\\"></div>"`;

exports[`Indicator component should have correct classes on different states 2`] = `"<div class=\\"indicator -success\\"></div>"`;
